### PR TITLE
Remove subprocess compatibility layer since Python 2.6 is no longer supported

### PR DESCRIPTION
**Pull request by Pratik151**
#4480 

Oops this got messed up. Will close this and make a new PR

### CODE REVIEW

**Comment by embray** [comment]
As I mentioned [here](https://github.com/astropy/astropy/pull/4473#issuecomment-171005589) you don't need to make a new PR.  You can fix your branch locally and then do a `git push --force origin` of the branch back to your remote branch.

**Comment by embray** [comment]
In this case, it looks like you tried to start this work from a branch off of a previous branch for a different issue, and then made various attempts to merge it back into master.

Since it looks like there's only one commit related to *this* issue it's easy.  First `git fetch upstream` to make sure your remote branches are up to date.  Then check out the branch you made for this issue:
```
$ git checkout remove-subprocess-comp
```
Reset the branch entirely to point to the last commit on upstream/master:
```
$ git reset --hard upstream/master
```
This throws away all work you did previously on that branch, and just resets it to be branching off the latest commit in upstream/master.

Now add back the commit you made for this change:
```
$ git cherry-pick a9b2016915c758a2953cf4a7e896f2b09979486a
```
And push back to your fork on GitHub:
```
$ git push --force origin remove-subprocess-comp
```
This updates the remote branch to be the same as your local branch of the same name again.  This will also automatically update the PR.

**Comment by MSeifert04** [comment]
Just a thought: I would have been very happy if https://github.com/astropy/astropy/pull/4483#issuecomment-171358998 would be on some of the ``contribute to astropy`` pages since this happened to me on one of my first pushs and I've seen others who had the same issue.

**Comment by embray** [comment]
@MSeifert04 See #4484

**Comment by Pratik151** [comment]
@astrofrog  can you review this when you have time :)

File: astropy/config/tests/test_configs.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
 from ...tests.helper import catch_warnings
 from ...extern import six
-from ...utils.compat import subprocess
+import subprocess
```
Comment (astrofrog) [inline astropy/config/tests/test_configs.py]: We usually follow the convention to have built-in modules at the top, so can you move this line to just under ``import sys``?

File: astropy/utils/tests/test_subprocess.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+def test_import_warning():
+    with catch_warnings() as w:
+        from ..compat import subprocess
+
+    assert len(w) == 1
+    assert w[0].category == AstropyDeprecationWarning
+    assert w[0].message.args[0] == ("astropy.utils.compat.subprocess is now deprecated "
+                                    "- use the Python subprocess module directly instead")
+
+
```
Comment (astrofrog) [inline astropy/utils/tests/test_subprocess.py]: Remove the two empty lines here

**Comment by astrofrog** [comment]
@Pratik151 - this is looking good, though see the two comments above. Another comment is that the commit message could be improved, but I can give instructions for that once the above comments are resolved.

**Comment by Pratik151** [comment]
@astrofrog  I resolved the issues which you commented.

Build is failing now because of 
<code>Error: Unsatisfiable package specifications.</code>

`Hint: the following packages conflict with each other:`
<code>  - numpy 1.9*</code>
<code>  - matplotlib 1.5|1.5.0*</code>
How can I resolve this?

**Comment by astrofrog** [comment]
@bsipocz - it looks like we should maybe not be pinning the Matplotlib version in ``ci-helpers`` because it will not work with old numpy versions. I wonder if we should just revert that change, or at least only pin it if the Numpy version is recent enough?

**Comment by astrofrog** [comment]
@bsipocz - ah wait, this is because we have ``NUMPY_VERSION`` as the default, and it remains that for the Sphinx build. Maybe we should actually do ``conda install matplotlib <= 1.5.0`` in ci-helpers instead of =1.5.0 in case other packages also have the same set-up? (basically, older than 1.5.0 is fine too)

**Comment by bsipocz** [comment]
@astrofrog - OK, I'll remove ``=1.5.0`` from ci-helpers. Don't like that we have to hard-wire it anyways as ``1.5.1`` works with python2.7, but hopefully the new bugfix of mpl will be out soon.

**Comment by bsipocz** [comment]
@astrofrog - ``ci-helpers`` is fixed, could you restart the failing builds?

**Comment by astrofrog** [comment]
@bsipocz - I've restarted the build - thanks!

**Comment by embray** [comment]
I think this looks good--any further comments @astrofrog ?

File: astropy/utils/tests/test_subprocess.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+def test_import_warning():
+    with catch_warnings() as w:
+        from ..compat import subprocess
+
+    assert len(w) == 1
+    assert w[0].category == AstropyDeprecationWarning
+    assert w[0].message.args[0] == ("astropy.utils.compat.subprocess is now deprecated "
+                                    "- use the Python subprocess module directly instead")
```
Comment (embray) [inline astropy/utils/tests/test_subprocess.py]: I really don't think this test is necessary.  All it's testing is that warnings work.  I can take it or leave it though.

Comment (astrofrog) [inline astropy/utils/tests/test_subprocess.py]: I agree it's not strictly necessary - I did it for the other removed compat modules though, so for consistency we can leave it (then remove after 1.2)

**Comment by astrofrog** [comment]
I merged this manually in 85bb53307ca9fa111767b8ebbca5a3866211632c since there was a conflict

**Comment by astrofrog** [comment]
Thanks @Pratik151!

**Comment by Pratik151** [comment]
@astrofrog Sorry didn't notice conflicts. (didn't checked this since few days). Thanks for doing it :)

**Pull request closed without merging**